ext/xmlreader/php_xmlreader.c

```diff
@@ -509,9 +509,8 @@ static void php_xmlreader_set_relaxng_schema(INTERNAL_FUNCTION_PARAMETERS, int t
 		}
 	}
 
-	php_error_docref(NULL, E_WARNING, "Unable to set schema. This must be set prior to reading or schema contains errors.");
-
-	RETURN_FALSE;
+	zend_throw_error(NULL, "Unable to set schema. This must be set prior to reading or schema contains errors.");
```

Comment:
"schema contains errors" doesn't sound like a promotable condition.

Reply:
I think this needs to be split into the error condition for `!intern || !intern->ptr` (can throw) and the one for retval != 0 (cannot throw).

Reply:
Thanks for the recommendation, I managed to separate them, and I added test coverage for both the warning and the exception.